javalib/src/main/scala/java/util/ArrayList.scala

```diff
@@ -0,0 +1,80 @@
+package java.util
+
+import scala.scalajs._
+import scala.collection.JavaConversions._
+
+class ArrayList[E] private (inner: js.Array[Box[E]]) extends AbstractList[E]
+                                                        with RandomAccess
+                                                        with Cloneable
+                                                        with Serializable {
+  self =>
+
+  def this(initialCapacity: Int) = {
+    this(new js.Array[Box[E]])
+    if (initialCapacity < 0)
+      throw new IllegalArgumentException
+  }
+
+  def this() =
+    this(10)
```

Comment:
This is unnecessary inefficient. It could call directly `this(new js.Array[Box[E]])`. Actually, I would make the overload taking `initialCapacity` call this one instead.